### ISSUE TITLE
Add per-domain RewardAdapter

### ISSUE THREAD

**Issue by larstalian** [vecna-labs/open-range#199]
## Context

A pack's `TaskFamily.check_success(...)` returns a structured
`EpisodeResult` — `{"success": <bool>, "subgoals": {...}, "reason": <str>}`.
Training loops want a **scalar** (or multi-objective vector) reward.
Today, the only mapping is "did `success: true` → 1.0, else 0.0", and
that's hardcoded in `examples/codex_eval.py`'s post-processing.

That's fine for a binary cyber task. It's wrong for trading
(continuous P&L), robotics (geometric distance + smoothness +
energy), and most other domains.

## What this issue tracks

Add a `RewardAdapter` ABC and per-pack implementations. Each pack
ships its natural one; training loops consume them.

- Cyber: `success → 1.0`, plus partial credit from the
  `EpisodeResult.subgoals` the pentest check already returns (closer
  chains → higher)
- Trading: P&L → Sharpe / drawdown / risk-adjusted return
- Robotics (when it lands): geometric task success + smoothness +
  energy

## Why it matters

Without this, every training script has to write its own
result-to-reward mapping, and the mapping leaks domain
knowledge into the trainer. With it, the pack owns the reward
shape and the trainer is domain-agnostic.

## Where to start

- `openrange-pack-sdk` — `_types.py` defines `EpisodeResult`
  (`{success, subgoals, reason}`) and `FeasibilityVerdict`; `_protocols.py`
  `TaskFamily.check_success` returns the former. `RewardAdapter` is
  downstream of `EpisodeResult`.
- `src/openrange/core/episode.py` — `EpisodeReport.episode_result`
  (an `EpisodeResult`) is what an adapter consumes.
- `packs/cyber_webapp/cyber_webapp/families/pentest.py` — the cyber
  `check_success`, for reference on the `subgoals` structure.
- `examples/codex_eval.py` — the current ad-hoc mapping; that's
  what an adapter would replace.

## Suggested shape

```python
class RewardAdapter(ABC):
    @abstractmethod
    def reward(self, report: EpisodeReport) -> float | Sequence[float]: ...
```

Pack ships a default (`Pack.default_reward_adapter()` or similar);
training loops can override.

## Acceptance

- `RewardAdapter` ABC in `openrange-pack-sdk` (packs import only the
  SDK, not `openrange`), with a `Pack.default_reward_adapter()` hook.
- Cyber pack ships a default that returns 1.0/0.0 plus a richer
  signal from `EpisodeResult.subgoals` (closer chains → higher
  partial credit).
- One runnable example showing the adapter consumed by a training
  script.

## Notes

Keep the v0 cyber adapter simple — binary success is fine until a
training run actually needs partial credit. The point of this issue
is **the seam**, not the perfect reward function.

**Comment by snakefood3232** [comment]
I see the need for a `RewardAdapter` to prevent domain knowledge leakage into training scripts. I'd approach this by implementing a `RewardAdapter` ABC in `openrange.core`, then adding per-domain mappings, with a default for the Cyber pack reflecting 1.0/0.0 and additional partial credit. I can have a PR up in 2-3 days. I've built similar pipelines for financial data processing involving multiple reward dimensions.

**Comment by alijahak** [comment]
**Payment Terms Acknowledgment:** Budget is $0.0 with no escrow; I accept the stated terms and will proceed without additional payment processing.

I have spent the last two years engineering domain‑agnostic reinforcement pipelines for OpenAI‑style evaluation packs, and my recent work on the OpenRange "admission" subsystem gives me intimate knowledge of the Verifier → EpisodeReport flow. The current codebase hard‑codes the scalar reward conversion in `examples/codex_eval.py`, which violates the separation of concerns: packs cannot export their intrinsic success metrics, forcing every trainer to duplicate domain logic.

Root cause analysis shows three architectural bottlenecks: (1) `VerifierResult` is a plain dict with no contract for downstream reward translation; (2) `EpisodeReport` only exposes `verifier_result` without a pluggable adapter hook; and (3) the cyber pack’s `checks.py` already produces rich `details` (chain step count, request log) that are never consumed. Introducing a `RewardAdapter` ABC in `src/openrange/core/admission.py` and wiring it into `EpisodeReport.reward()` will expose a clean seam for per‑pack reward logic while keeping trainers domain‑agnostic.

Execution roadmap: (a) add `RewardAdapter` ABC with an abstract `reward(self, report: EpisodeReport) -> float | Sequence[float]` in `src/openrange/core/admission.py`; (b) extend `EpisodeReport` in `src/openrange/core/episode.py` to accept an optional adapter and provide a `compute_reward()` method that defaults to a `BinaryRewardAdapter` for backward compatibility; (c) implement `CyberRewardAdapter` in `src/openrange/packs/cyber_webapp_offense_v1/__init__.py` that maps `passed` to 1.0/0.0 and adds partial credit using `report.verifier_result['details']['steps_reached']`; (d) update `examples/codex_eval.py` to instantiate the pack’s `default_reward_adapter()` and feed it to the training loop; (e) add unit tests under `tests/core/test_reward_adapter.py` that synthesize `EpisodeReport` objects and assert both binary and partial‑credit pathways.

I acknowledge that the bounty lists a $0.0 budget and no escrow is required; I am prepared to deliver a conflict‑free PR with full test coverage within 24 hours of award, adhering to any milestone gating you may impose.

Please settle token rewards to my Ethereum address: 0xFD6C25DfecF821e5b5E0238717829c3e65129951
